File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode(val=0)
        curr = head
        prev = dummy

        dummy.next = head

        while curr:
            if curr.val == val:
                prev.next = prev.next.next
            else:
                prev = prev.next
            
            curr = curr.next
        
        return dummy.next  

File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let curr = head;
    let dummy = new ListNode();
    let prev = dummy;

    dummy.next = head;
    while(curr){
        if(curr.val == val){
            prev.next = prev.next.next;
            
        } else {
            prev = prev.next;
        }
        curr = curr.next;
    }
    return dummy.next;
};